yarnitor/background/worker.py

```diff
@@ -361,22 +520,42 @@ def run_update(self):
         # https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators
         self.state["refresh-datetime"] = datetime.datetime.utcnow().isoformat() + 'Z'
         self.redis_client.set(YARN_STATUS_KEY, json.dumps(self.state))
+        logger.info("Done updating metrics from YARN")
+
+    def loop(self, sleep_time):
+        """Executes the `run_update` message in a loop that catches and logs
+        all exceptions indefinitely.
 
-    def loop(self):
+        Parameters
+        ----------
+        sleep_time: float
+            Time to sleep after executing `run_update`
+        """
         while True:
             try:
                 self.run_update()
             except Exception:
                 logger.exception('Unknown exception while updating')
-            time.sleep(self.sleep_time)
+            time.sleep(sleep_time)
 
 
 def main():
-    host, port = os.getenv('REDIS_ENDPOINT', "localhost:6379").split(":")
+    """Creates a redis client, a YARN ResourceManager REST API client, and a YARN
+    poller that puts information about the YARN cluster and its applications into
+    redis on a timed interval.
+    """
+    log_level = os.getenv('LOG_LEVEL', 'INFO')
+    logging.basicConfig(level=getattr(logging, log_level))
+
+    host, port = os.environ['REDIS_ENDPOINT'].split(":")
     redis_client = redis.StrictRedis(host=host, port=port)
+    yarn_handler = YARNHandler(os.environ['YARN_ENDPOINT'])
 
-    ym = YARNModel(redis_client)
-    ym.loop()
+    ym = YARNPoller(redis_client, yarn_handler)
+    ym.register_handler("SPARK", SparkHandler)
+    ym.register_handler("MAPREDUCE", MapredHandler)
+    ym.register_handler("MAPRED", MapredHandler)
```

Comment:
Why is there `MAPREDUCE` and `MAPRED`?

Reply:
Both are seen depending on the version of MapReduce the job is running.